### ISSUE TITLE
NextRouter was not mounted (in the pages directory)

### ISSUE THREAD

**Issue by andresmarpz** [vercel/next.js#54244]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: linux
      Arch: x64
      Version: #85~20.04.1-Ubuntu SMP Mon Jul 17 09:42:39 UTC 2023
    Binaries:
      Node: 18.17.0
      npm: 9.6.7
      Yarn: 1.22.19
      pnpm: N/A
    Relevant Packages:
      next: 13.4.18-canary.0
      eslint-config-next: 13.4.18
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/andresmarpz/next-router-not-mounted

### To Reproduce

Navigate to localhost:3000

### Describe the Bug

Basically we can't use `useRouter` outside of the context of Next.js (for example, on `getServerSideProps` with `renderToString`). This is a problem because there are dependencies that still depend on it (ie. Algolia Instant Search).

### Expected Behavior

We should be able to use `useRouter` inside getServerSideProps with `renderToString` as we can in next@12 (in the pages directory still!). This is a huge breaking change otherwise, if there wasn't any update to the pages directory, it should still be OK to leave as-is

**Comment by andresmarpz** [comment]
Actually, I'm yet to try updating the import to `next/compat/router`, but that looks like what I was looking for. Referred to https://github.com/vercel/next.js/pull/42502. Since docs at [next-router-not-mounted](https://nextjs.org/docs/messages/next-router-not-mounted) don't mention this, I opened a pull request (https://github.com/vercel/next.js/pull/54317) to add some context there for users that click the error message.

**Comment by balazsorban44** [comment]
This has not been the intended usage starting with version 13, but to keep compatibility, https://github.com/vercel/next.js/pull/42502 introduced a `next/compat/router` version that should work for this case.

```diff
- import { useRouter } from "next/router";
+ import { useRouter } from "next/compat/router";
```
